chronicles/log_output.nim

```diff
@@ -72,6 +73,13 @@ type
                    colors: static[ColorScheme]] = object
     output*: Output
 
+  TextBlockBRecord*[Output;
```

Comment:
You can drop all Record types in favour of a single generic `Record` type. It will have a `Format` parameter that will be either `Json`, `TextBlockLog` or `TextLineLog`.

From the `Format` parameter, you can get the writer type with `WriterType(Format)`

Reply:
Getting a generic Record was fairly straightforward, but a generic Writer was been difficult since I'd want inheritance-like behavior. See question in PR thread.